src/livekit/useLiveKit.ts

```diff
@@ -101,6 +101,11 @@ export function useLiveKit(
   // block audio from being enabled until the connection is finished.
   const [blockAudio, setBlockAudio] = useState(true);
 
+  // Store if audio/video are currently updating. If to prohibit unnecessary calls
+  // to setMicrophoneEnabled/setCameraEnabled
+  const [audioMuteUpdating, setAudioMuteUpdating] = useState(false);
+  const [videoMuteUpdating, setVideoMuteUpdating] = useState(false);
```

Comment:
I suspect you probably just want `useRef` for this, otherwise setting them is going to cause a re-render.